wakuv2/waku.go

```diff
@@ -995,13 +996,19 @@ func (w *Waku) UnsubscribeMany(ids []string) error {
 	return nil
 }
 
+func (w *Waku) SetSendFailures(value bool) {
+	w.settings.SendFailures = value
+}
+
 func (w *Waku) broadcast() {
 	for {
 		select {
 		case envelope := <-w.sendQueue:
 			var err error
 			logger := w.logger.With(zap.String("envelopeHash", hexutil.Encode(envelope.Hash())), zap.String("pubsubTopic", envelope.PubsubTopic()), zap.String("contentTopic", envelope.Message().ContentTopic), zap.Int64("timestamp", envelope.Message().Timestamp))
-			if w.settings.LightClient {
+			if w.settings.SendFailures {
+				err = errors.New("Test send failure")
```

Comment:
How about change `SendFailures` to `SkipPublishToTopic` which may feel less test specific, and add a short comment explain it mostly used in test scenarios?
Wondering if `err` is needed here, I haven't look into how `EventEnvelopeExpired` is used.

Reply:
Good idea, thanks, renamed. The error value at least is useful in tests, it's visible in the output.